documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,292 @@
+
+The cmdlets currently found in the `Az.Resources` module that act upon service principals are the following:
+
+- [`Get-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzADServicePrincipal.md)
+- [`Get-AzADSpCredential`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzADSpCredential.md)
+- [`New-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzADServicePrincipal.md)
+- [`New-AzADSpCredential`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzADSpCredential.md)
+- [`Remove-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Remove-AzADServicePrincipal.md)
+- [`Remove-AzADSpCredential`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Remove-AzADSpCredential.md)
+- [`Update-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Update-AzADServicePrincipal.md)
+
+#### User Scenarios
+
+A common scenario is for a user to create a service principal in order to give it specific permissions over a subscription, resource group, or resource in order to automate tasks with the service principal. When the user creates a service principal using the `New-AzADServicePrincipal` cmdlet, there is no way using the Graph SDK to also give the service principal the desired permissions over a given scope; the user would need to go through the portal to manually create the permissions. One of the scenarios we added to this cmdlet was to give users the ability to also create a role assignment for the service principal within this cmdlet using new `-Role` and `-Scope` parameters. If the user provided a value for either parameter, once the service principal was created, a [call to the authorization SDK](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L346-L380) would be made to assign the specified role to the service principal over the given scope. If neither parameter was provided, or the `-SkipAssignment` switch was set, then a role assignment would not be created. If only `-Role` was provided, `-Scope` would be [set to the user's current subscription](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L252-L256), and if only `-Scope` was provided, `-Role` would be [set to the "Contributor"](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L260-L264).
```

Comment:
A common scenario is for a user to create a service principal in order to automate Azure management tasks. When a Service Principal is used to automate management tasks, it must have a credential, and it must be given permissions to manage the resources that are part of its task.  The `New-AzADServicePricnipal` cmdlet allows the user to create a service principal, give it secure credentials, and allow the service principal the appropriate level of access in Azure through issuing a single cmdlet.